### ISSUE TITLE
Wrapping a TFGPT2LMHeadModel With GradientAccumulateModel

### ISSUE THREAD

**Issue by kazunator** [andreped/GradientAccumulator#33]
I get an error when I try to wrap my gpt2 model with the ga wrapper. Here is a screenshot:
![image](https://user-images.githubusercontent.com/22581232/215907453-1d3f4de9-4bd2-4886-85c4-b4d43c321dee.png)

**Comment by 0syrys** [comment]
Hi @kazunator,

I have the same issue with my Huggingface TF model
Take a look [here ](https://github.com/andreped/GradientAccumulator/issues/30#issuecomment-1411675056) and see if something similar works for your gpt2 as well

**Comment by andreped** [comment]
Yes, see @0syrys's comment [here](https://github.com/andreped/GradientAccumulator/issues/30#issuecomment-1411675056) for how he solved that. And please, let me know whether it was successful for your application.

**Comment by andreped** [comment]
@0syrys made a notebook which I have added to the main branch (see [here](https://github.com/andreped/GradientAccumulator/blob/main/notebooks/GA_for_HuggingFace_TF_models.ipynb)). I will update the README regarding how to use `gradient-accumulator` with HuggingFace TF models next.

Closing this issue for now.

**Comment by andreped** [comment]
@0syrys and @kazunator If you are using gradient accumulation in distributed training, I have now added multi-GPU support.

It should work out-of-the-box with the optimizer wrapper, whereas the model wrapper has experiemental SGD support.
